src/apollo/queries.js

```diff
@@ -1,5 +1,124 @@
 import gql from 'graphql-tag';
 
+export const UNISWAP_PAIRS_ID_QUERY = gql`
+  query pairs {
+    pairs(first: 200, orderBy: trackedReserveETH, orderDirection: desc) {
+      id
+    }
+  }
+`;
+
+export const UNISWAP_PAIR_DATA_QUERY = (pairAddress, block) => {
+  const queryString = `
+    fragment PairFields on Pair {
+      id
+      txCount
+      token0 {
+        id
+        symbol
+        name
+        totalLiquidity
+        derivedETH
+      }
+      token1 {
+        id
+        symbol
+        name
+        totalLiquidity
+        derivedETH
+      }
+      reserve0
+      reserve1
+      reserveUSD
+      totalSupply
+      trackedReserveETH
+      reserveETH
+      volumeUSD
+      untrackedVolumeUSD
+      token0Price
+      token1Price
+      createdAtTimestamp
+  }
+  query pairs {
+    pairs(${
+      block ? `block: {number: ${block}}` : ``
+    } where: { id: "${pairAddress}"} ) {
+      ...PairFields
+    }
+  }`;
+  return gql`
+    ${queryString}
+  `;
+};
+
+export const UNISWAP_PAIRS_BULK_QUERY = gql`
+  fragment PairFields on Pair {
+    id
+    txCount
+    token0 {
+      id
+      symbol
+      name
+      totalLiquidity
+      derivedETH
+    }
+    token1 {
+      id
+      symbol
+      name
+      totalLiquidity
+      derivedETH
+    }
+    reserve0
+    reserve1
+    reserveUSD
+    totalSupply
+    trackedReserveETH
+    reserveETH
+    volumeUSD
+    untrackedVolumeUSD
+    token0Price
+    token1Price
+    createdAtTimestamp
+  }
+  query pairs($allPairs: [Bytes]!) {
+    pairs(
+      where: { id_in: $allPairs }
+      orderBy: trackedReserveETH
+      orderDirection: desc
+    ) {
+      ...PairFields
+    }
+  }
+`;
+
+export const UNISWAP_PAIRS_HISTORICAL_BULK_QUERY = (block, pairs) => {
+  let pairsString = `[`;
```

Comment:
Note for later: we should pass in pairs as argument [String]! and see if there's any reason why Uniswap did the queryString this way